### PR TITLE
[Doc] Align with verl

### PR DESCRIPTION
**Pull request by hiyuchang**
## Description

As the title says.

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has passed all tests
- [ ]  Docstrings have been added/updated in Google Style
- [ ]  Documentation has been updated
- [ ]  Code is ready for review

### CODE REVIEW

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+Trinity-RFT divides massive parameters of reinforcement fine-tuning into several parts according to their functions, e.g., `algorithm`, `model`, `buffer`, `explorer`, `trainer`, `monitor`, and `synchronizer`. Such detailed and flexible configuration of parameters enables users to customize the training process.
+Please refer to the [documentation](https://modelscope.github.io/Trinity-RFT/en/main/tutorial/trinity_configs.html) for the detailed parameter configuration of Trinity-RFT.
+
+In the following, we show how to map the parameters in veRL to the ones in Trinity-RFT. The core parameters in veRL are divided into these categories: `algorithm`, `data`, `actor_rollout_ref`, `critic`, `reward_model`, and `trainer`.
+
+```{note}
+To match the default training setup of veRL, we set `synchronizer.sync_style=fixed` and `synchronizer.sync_offset=1` in Trinity-RFT.
+```
+
+**Algorithm**
```
Comment (pan-x-c) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: -> ### Algorithm

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+| `algorithm.lam` | `advantage_fn_args.lam` | Along with `advantage_fn: ppo` |
+| `algorithm.use_kl_in_reward` | `algorithm.kl_penalty_fn` | Disable KL in reward by setting `algorithm.kl_penalty_fn=none` |
+| `algorithm.kl_penalty` | `algorithm.kl_penalty_fn` | Choose from `k2`, `low_var_kl`, etc |
+| `algorithm.kl_ctrl.kl_coef` | `algorithm.kl_penalty_fn_args.kl_coef` | - |
+
+💡 Detailed explanation:
+* Before using args of advantage function or policy loss function, a good practice is to check the source code to ensure these parameters can be processed by the corresponding function properly.
+
+
+**Data**
```
Comment (pan-x-c) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: Same as above

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+```{note}
+To match the default training setup of veRL, we set `synchronizer.sync_style=fixed` and `synchronizer.sync_offset=1` in Trinity-RFT.
+```
+
+**Algorithm**
+
+| veRL | Trinity-RFT | Note |
+|:-----|:-----|:-----|
+| `algorithm.adv_estimator` | `algorithm.advantage_fn` | Pass parameters with `algorithm.advantage_fn_args` |
+| `algorithm.gamma` | `advantage_fn_args.gamma` | Along with `advantage_fn: ppo` |
```
Comment (pan-x-c) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: not only `ppo` use this parameter

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 4]
```diff
@@ -0,0 +1,462 @@
+# Aligning with veRL
+
+This guide provides a guide for users familiar with [veRL](https://github.com/volcengine/verl) to align the parameters in Trinity-RFT with the ones in veRL.
+
+
+## Parameter Mapping
```
Comment (pan-x-c) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md:8]: Add a metrics mapping section.
Explain
- why there are 2 experiments (`-explorer`, `-trainer`) for one run.
- why some metrics `actor/ppo_kl`, `actor/pg_clipfrac` are different from VeRL.

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+
+💡 Detailed explanation:
+* Before using args of advantage function or policy loss function, a good practice is to check the source code to ensure these parameters can be processed by the corresponding function properly.
+
+
+### Data
+
+| veRL | Trinity-RFT | Note |
+|:-----|:-----|:-----|
+| `data.train_files` | `buffer.explorer_input.taskset.path` | - |
```
Comment (chenyushuo) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]:
```suggestion
| `data.train_files` | `buffer.explorer_input.taskset.path` or `buffer.explorer_input.tasksets[i].path` | - |
```

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+```
+
+For advanced training configuration of veRL you can set these up in the field of `trainer.trainer_config`. For example,`actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu` in veRL is equivalent to `trainer.trainer_config.actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu` in Trinity-RFT. If you want to setup the parameters in the `trainer.trainer_config` dictionary, please read the source code in `trinity/common/verl_config.py` carefully!
+
+
+| veRL | Trinity-RFT | Note |
+|:-----|:-----|:-----|
+| `actor_rollout_ref.model.path` | `model.model_path` | - |
+| `actor_rollout_ref.actor.optim` | `algorithm.optimizer` | Such as `lr` and `weight_decay` |
+| `actor_rollout_ref.rollout.n` | `algorithm.repeat_times` | Task-specific: `taskset.repeat_times` |
```
Comment (chenyushuo) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: We can not set `taskset.repeat_times` for train data.

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+| `actor_rollout_ref.actor.ppo_mini_batch_size` | `buffer.batch_size` | The number of tasks to be explored in each exploration step |
+| `actor_rollout_ref.actor.use_dynamic_bsz` | `trainer.use_dynamic_bsz` | - |
+| `actor_rollout_ref.actor.ppo_max_token_len_per_gpu` | `trainer.max_token_len_per_gpu` | - |
+| `actor_rollout_ref.actor.ulysses_sequence_parallel_size` | `trainer.ulysses_sequence_parallel_size` | The sequence parallel size for the actor |
+| `actor_rollout_ref.actor.grad_clip` | `trainer.grad_clip` | The gradient clip value for the actor |
+| `actor_rollout_ref.actor.use_kl_loss` | `algorithm.kl_loss_fn` | If set to `none`, the KL divergence loss will not be computed |
+| `actor_rollout_ref.rollout.gpu_memory_utilization` | `explorer.rollout_model.gpu_memory_utilization` | - |
+| `actor_rollout_ref.rollout.temperature` | `model.temperature` | Can be task-specific |
```
Comment (chenyushuo) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]:
```suggestion
| `actor_rollout_ref.rollout.temperature` | `model.temperature` | Can be taskset-specific, like `buffer.explorer_input.taskset.rollout_args.temperature` |
```

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+| `trainer.logger` | `monitor.monitor_type` | Support a chosen type and (no need to set) `console` |
+| `trainer.project_name` | `project` | - |
+| `trainer.experiment_name` | `name` | - |
+| `trainer.default_local_dir` | `checkpoint_root_dir` | Checkpoint is saved in `<checkpoint_root_dir>/<project>/<name>/` |
+| `trainer.n_gpus_per_node` | `cluster.gpu_per_node` | - |
+| `trainer.nnodes` | `cluster.node_num` | - |
+| `trainer.save_freq` | `trainer.save_interval` | - |
+| `trainer.test_freq` | `explorer.eval_interval` | - |
+| `trainer.total_epochs` | `buffer.total_epochs` | - |
+| `trainer.total_training_steps` | `buffer.total_steps` | If not None, `buffer.total_epochs` will be ignored |
```
Comment (chenyushuo) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]:
```suggestion
| `trainer.total_training_steps` | `buffer.total_steps` and `trainer.total_steps` | If not None, `buffer.total_epochs` will be ignored |
```

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+
+## Parameter Mapping
+
+Trinity-RFT divides massive parameters of reinforcement fine-tuning into several parts according to their functions, e.g., `algorithm`, `model`, `buffer`, `explorer`, `trainer`, `monitor`, and `synchronizer`. Such detailed and flexible configuration of parameters enables users to customize the training process.
+Please refer to the [documentation](https://modelscope.github.io/Trinity-RFT/en/main/tutorial/trinity_configs.html) for the detailed parameter configuration of Trinity-RFT.
+
+In the following, we show how to map the parameters in veRL to the ones in Trinity-RFT. The core parameters in veRL are divided into these categories: `algorithm`, `data`, `actor_rollout_ref`, `critic`, `reward_model`, and `trainer`.
+
+```{note}
+To match the default training setup of veRL, we set `synchronizer.sync_style=fixed` and `synchronizer.sync_offset=1` in Trinity-RFT.
```
Comment (yanxi-chen) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: Should it be `sync_offset=0`?

File: docs/sphinx_doc/source_zh/tutorial/align_with_verl.md [hunk 10]
```diff
@@ -0,0 +1,492 @@
...
+| `actor_rollout_ref.rollout.temperature` | `model.temperature` | 可以是taskset-specific，例如 `buffer.explorer_input.taskset.rollout_args.temperature` |
+| `actor_rollout_ref.rollout.top_p` | `model.top_p` | 可以是taskset-specific |
+| `actor_rollout_ref.rollout.top_k` | `model.top_k` | 可以是taskset-specific |
+| `actor_rollout_ref.rollout.tensor_model_parallel_size` | `explorer.rollout_model.tensor_parallel_size` | - |
+| `actor_rollout_ref.rollout.val_kwargs` | `buffer.explorer_input.eval_tasksets[i]` | Taskset-specific |
+| `critic.model.path` | `model.critic_model_path` | 默认为 `model.model_path` |
+
+💡 详细说明：
+
+* 注释 `可以是taskset-specifc`（以 `temperature` 为例）意味着您可以为所有任务集设置 `model.temperature`，或者在 `buffer.explorer_input.taskset.rollout_args.temperature` 或 `buffer.explorer_input.eval_tasksets[i].rollout_args.temperature` 中为每个任务设置不同的值。具体示例如下：
```
Comment (ShenQianli) [inline docs/sphinx_doc/source_zh/tutorial/align_with_verl.md]: specifc -> specific

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 11]
```diff
@@ -0,0 +1,493 @@
+# Aligning with veRL
+
+This guide provides a guide for users familiar with [veRL](https://github.com/volcengine/verl) to align the parameters and metrics in Trinity-RFT with the ones in veRL.
```
Comment (HYLcool) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: "This **guide** provides a **guide** for ..."
Kind of duplicate words. Maybe some other options are better:
1. This guide provides guidance for ...
2. This guide helps users familiar with veRL to ...
3. ......

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 12]
```diff
@@ -0,0 +1,493 @@
...
+
+* If you want to filter the overlong prompts, you can set `model.enable_prompt_truncation=True` in Trinity-RFT. In this case, the corresponding experiences will not be counted in loss computation, and thus `truncation` side does not matter anymore.
+
+
+### Actor, Rollout, and Critic
+
+This section includes the parameters for the actor and the rollout. For easy understanding, you may think the actor in veRL (`actor_rollout_ref.actor`) as the trainer in Trinity-RFT (`trainer`), and the rollout (`actor_rollout_ref.rollout`) as the explorer (`explorer.rollout_model`).
+
+```{note}
+Any parameter in `actor_rollout_ref.rollout` in Trinity is not effective; please set them in other fields properly.
```
Comment (HYLcool) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md]: in Trinity --> in Trinity-RFT

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 13]
```diff
@@ -0,0 +1,493 @@
...
+This guide provides a guide for users familiar with [veRL](https://github.com/volcengine/verl) to align the parameters and metrics in Trinity-RFT with the ones in veRL.
+
+
+## Parameter Mapping
+
+Trinity-RFT divides massive parameters of reinforcement fine-tuning into several parts according to their functions, e.g., `algorithm`, `model`, `buffer`, `explorer`, `trainer`, `monitor`, and `synchronizer`. Such detailed and flexible configuration of parameters enables users to customize the training process.
+Please refer to the [documentation](https://modelscope.github.io/Trinity-RFT/en/main/tutorial/trinity_configs.html) for the detailed parameter configuration of Trinity-RFT.
+
+In the following, we show how to map the parameters in veRL to the ones in Trinity-RFT. The core parameters in veRL are divided into these categories: `algorithm`, `data`, `actor_rollout_ref`, `critic`, `reward_model`, and `trainer`.
+
```
Comment (ShenQianli) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md:26]: Users who are familiar with veRL but not with Trinity—the primary audience for this document—may wonder:
	1.	how the Trinity modules relate (roughly) to the corresponding veRL modules, and
	2.	why Trinity introduces separate buffer, explorer, and synchronizer components.

Providing brief answers to these questions here would help readers follow the rest of the document more smoothly.

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 14]
```diff
@@ -0,0 +1,493 @@
...
+### Data
+
+| veRL | Trinity-RFT | Note |
+|:-----|:-----|:-----|
+| `data.train_files` | `buffer.explorer_input.taskset.path` or `buffer.explorer_input.tasksets[i].path` | - |
+| `data.val_files` | `buffer.explorer_input.eval_tasksets[i].path` | - |
+| `data.prompt_key` | `buffer.explorer_input.taskset.format.prompt_key`| Task-specific |
+| `data.response_key` | `buffer.explorer_input.taskset.format.response_key`| Task-specific |
+| `data.train_batch_size` | `buffer.batch_size` * `synchronizer.sync_interval` | The number of tasks to be explored |
```
Comment (ShenQianli) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md:57]: It may be helpful to highlight parameters such as train_batch_size which are critical for performance alignment.

File: docs/sphinx_doc/source/tutorial/align_with_verl.md [hunk 15]
```diff
@@ -0,0 +1,493 @@
+# Aligning with veRL
+
+This guide provides a guide for users familiar with [veRL](https://github.com/volcengine/verl) to align the parameters and metrics in Trinity-RFT with the ones in veRL.
+
+
+## Parameter Mapping
```
Comment (ShenQianli) [inline docs/sphinx_doc/source/tutorial/align_with_verl.md:8]: Do you think it would be helpful to provide guidance on setting the `engine_num`s (and their corresponding `batch_size`s)? This might be new for veRL users.
